File: Lab_2/task_1/1.cs
using System;
using System.Text;

class Program
{
    static string CompressString(string s)
    {
        StringBuilder compressed = new StringBuilder();
        int count = 1;

        for (int i = 1; i <= s.Length; i++)
        {
            if (i < s.Length && s[i] == s[i - 1])
            {
                count++;
            }
            else
            {
                compressed.Append(s[i - 1]).Append(count);
                count = 1;
            }
        }

        if (compressed.Length < s.Length)
        {
            return compressed.ToString();
        }
        else
        {
            Console.WriteLine("Сжатие не уменьшило строку. Возвращена исходная строка.");
            return s;
        }
    }

    static void Main()
    {
        Console.Write("Введите строку: ");
        string s = Console.ReadLine();
        Console.WriteLine("Результат: " + CompressString(s));
    }
}

File: Lab_2/task_1/1.kt
fun compressString(s: String): String {
    val compressed = StringBuilder()
    var count = 1

    for (i in 1..s.length) {
        if (i < s.length && s[i] == s[i - 1]) {
            count++
        } else {
            compressed.append(s[i - 1]).append(count)
            count = 1
        }
    }

    return if (compressed.length < s.length) {
        compressed.toString()
    } else {
        println("Сжатие не уменьшило строку. Возвращена исходная строка.")
        s
    }
}

fun main() {
    print("Введите строку: ")
    val s = readLine() ?: ""
    println("Результат: ${compressString(s)}")
}
